CMakeLists.txt

```diff
@@ -18,49 +47,333 @@ set(TANGO_HOST $ENV{TANGO_HOST})
+install(FILES $ENV{ZMQ_BASE}/lib/Debug/libzmq-v140-mt-gd-4_1_7.lib DESTINATION lib COMPONENT static)
+install(FILES $ENV{ZMQ_BASE}/bin/Debug/libzmq-v140-mt-gd-4_1_7.dll DESTINATION bin COMPONENT dynamic)
+else()
+install(FILES $ENV{OMNI_BASE}/bin/x86_win32/omniORB421_vc14_rt.dll DESTINATION bin COMPONENT dynamic)
+install(FILES $ENV{OMNI_BASE}/bin/x86_win32/omniDynamic421_vc14_rt.dll DESTINATION bin COMPONENT dynamic)
+install(FILES $ENV{OMNI_BASE}/bin/x86_win32/omnithread40_vc14_rt.dll DESTINATION bin COMPONENT dynamic)
+install(FILES $ENV{OMNI_BASE}/bin/x86_win32/COS421_vc14_rt.dll DESTINATION bin COMPONENT dynamic)
+install(FILES $ENV{ZMQ_BASE}/lib/Release/libzmq-v140-mt-4_1_7.lib DESTINATION lib COMPONENT static)
+install(FILES $ENV{ZMQ_BASE}/bin/Release/libzmq-v140-mt-4_1_7.dll DESTINATION bin COMPONENT dynamic)
+endif()
+ENDIF()
+#pthreads
+IF(CMAKE_CL_64)
+install(FILES $ENV{PTHREAD_WIN}/Pre-built.2/lib/x64/pthreadVC2.lib DESTINATION lib COMPONENT static)
+install(FILES $ENV{PTHREAD_WIN}/Pre-built.2/dll/x64/pthreadVC2.dll DESTINATION bin COMPONENT dynamic)
+ELSE()
+install(FILES $ENV{PTHREAD_WIN}/Pre-built.2/lib/x86/pthreadVC2.lib DESTINATION lib COMPONENT static)
+install(FILES $ENV{PTHREAD_WIN}/Pre-built.2/dll/x86/pthreadVC2.dll DESTINATION bin COMPONENT dynamic)
+ENDIF()
 
+
+configure_file(tango.pc.cmake tango.pc @ONLY)
+
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tango.pc"
+        DESTINATION include/pkgconfig COMPONENT headers)
+ELSE()
+install(TARGETS tango LIBRARY DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}")
 configure_file(tango.pc.cmake tango.pc @ONLY)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tango.pc"
         DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig")
+ENDIF()
+
+IF (WIN32)
+    IF(MSVC90)
+      if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        IF(CMAKE_CL_64)
+          set(CPACK_PACKAGE_FILE_NAME libtango_${LIBRARY_VERSION}_v90_x64_debug)
+        ELSE()
+          set(CPACK_PACKAGE_FILE_NAME libtango_${LIBRARY_VERSION}_v90_x86_debug)
+        ENDIF()
+      else()
+        IF(CMAKE_CL_64)
+          set(CPACK_PACKAGE_FILE_NAME libtango_${LIBRARY_VERSION}_v90_x64)
+        ELSE()
+          set(CPACK_PACKAGE_FILE_NAME libtango_${LIBRARY_VERSION}_v90_x86)
+        ENDIF()
+      endif(CMAKE_BUILD_TYPE STREQUAL "Debug")
+    ELSE()
+      IF(CMAKE_CL_64)
+        if (CMAKE_BUILD_TYPE STREQUAL "Debug")  
+          set(CPACK_PACKAGE_FILE_NAME libtango_${LIBRARY_VERSION}_${CMAKE_VS_PLATFORM_TOOLSET}_x64_debug)
+        else()
+          set(CPACK_PACKAGE_FILE_NAME libtango_${LIBRARY_VERSION}_${CMAKE_VS_PLATFORM_TOOLSET}_x64)
+        endif()
+      ELSE()
+        if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+          set(CPACK_PACKAGE_FILE_NAME libtango_${LIBRARY_VERSION}_${CMAKE_VS_PLATFORM_TOOLSET}_x86_debug)
+        else()
+          set(CPACK_PACKAGE_FILE_NAME libtango_${LIBRARY_VERSION}_${CMAKE_VS_PLATFORM_TOOLSET}_x86)
+        endif()
+      ENDIF()
+    ENDIF(MSVC90)
+  #
```

Comment:
Probably it would be better to move all cpack related stuff to a dedicated file and then include it here. Just to keep thing clean